app/models/course_reserve.rb

```diff
@@ -1,35 +1,8 @@
 # frozen_string_literal: true
 
 # The store of course reserve data from courses.json
-class CourseReserve
-  FolioCourseInfo = Data.define(:id, :course_number, :name, :instructor)
-
-  class Repository
-    include Singleton
-
-    def all
-      @all ||= Folio::Types.courses.map do |id, v|
-        FolioCourseInfo.new(id:, course_number: v['courseNumber'],
-                            name: v['name'],
-                            instructor: v.dig('courseListingObject', 'instructorObjects').pluck('name'))
-      end
-    end
-
-    def find(*ids)
-      if ids.length == 1
-        id = ids.first
-        all.find { |course| course.id == id }
-      else
-        all.select { |course| ids.include?(course.id) }
-      end
-    end
-  end
-
-  def self.all
-    Repository.instance.all
-  end
-
-  def self.find(*id)
-    Repository.instance.find(*id)
-  end
+class CourseReserve < ApplicationRecord
+  serialize :instructors, coder: YAML, type: Array
```

Comment:
Why YAML and not JSON?

Reply:
No reason except I needed a coder and that was one I saw existed. I can change to JSON.

Reply:
I ran into some issues with JSON - requires a little more review so leaving it for now for this PR but will revisit separately.